Find a functionally equivalent code:android.view.Surface.copyFrom:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * Copy another surface to this one.  This surface now holds a reference
 * to the same data as the original surface, and is -not- the owner.
 * This is for use by the window manager when returning a window surface
 * back from a client, converting it from the representation being managed
 * by the window manager to the representation the client uses to draw
 * in to it.
 * @hide
 */

@@ -2,12 +2,12 @@
     if (other == null) {
         throw new IllegalArgumentException("other must not be null");
     }
-    int surfaceControlPtr = other.mNativeObject;
+    long surfaceControlPtr = other.mNativeObject;
     if (surfaceControlPtr == 0) {
         throw new NullPointerException("SurfaceControl native object is null. Are you using a released SurfaceControl?");
     }
-    int newNativeObject = nativeCreateFromSurfaceControl(surfaceControlPtr);
+    long newNativeObject = nativeCreateFromSurfaceControl(surfaceControlPtr);
     synchronized (mLock) {
         if (mNativeObject != 0) {
             nativeRelease(mNativeObject);

